### PR TITLE
EN-12514: github community standards

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
added code of conduct, PR template, contributing guide, and also refactored issue templates

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+If you are unfamiliar with the workflow of contributing to github, you can refer to this [this article](https://github.com/firstcontributions/first-contributions/blob/master/README.md)
+
+## Fork & clone this repository
+
+The development should happen in a personal fork, cloned on the local machine.
+
+## Use development branch
+
+External contributions should happen against the `development` branch.
```
Comment (iulianpascalau) [inline .github/CONTRIBUTING.md:12]: :+1:

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Other branches may be used by the team members in accordance to internal decisions.
+
+## Use linter
+
+Make sure the code is well-formatted and aligned before opening a PR. Some linters such `gofmt` can be used, or code style
+can be proactively checked while written code when using an IDE.
```
Comment (iulianpascalau) [inline .github/CONTRIBUTING.md:19]: writing? + comma before when

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+## Writing tests
+
+In case of adding a new feature, make sure you follow these guidelines:
+- Attempt to write some unit tests which verify the component in isolation and mock the external components around it
```
Comment (iulianpascalau) [inline .github/CONTRIBUTING.md:24]: start all list entries with lower case, end them with `;` except the last one which ends with `.`

Valid for all lists

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+- If the component relies on many other external components which are difficult to mock, write integrations tests for it
```
Comment (iulianpascalau) [inline .github/CONTRIBUTING.md:25]: Please add:
```
Although, this usually means that further decoupling using interfaces is required. Also, the component might needed to be split in smaller, specialized sub-components.
```

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (raduchis) [inline .github/CONTRIBUTING.md:24]: Write unit tests which.... remove **attempt** and **some**

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+- Try to cover both expected functionality but also verify that the code fails in expected ways
```
Comment (raduchis) [inline .github/CONTRIBUTING.md:26]: Remove **try to**: Cover both expected functionality...

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+In case of fixing a bug:
+- Try to first write the unit test that exposes the bug, then fix it and have a test that will make sure the specific situation will always be tested in the future
+
+In both cases, make sure that the new code has a good code coverage (ideally, 100% of the new lines are covered) and also try to cover edge-cases.
+
```
Comment (iulianpascalau) [inline .github/CONTRIBUTING.md:32]: I think we should add:
```
Please take care of the severity of the bug. Pushing directly a fix of an undiscovered, critical bug or a bug that can be exploited in such a way that could create damage to the blockchain, loss of funds, state alteration beyond repair, and so on, can be exploited before the team can take any action. Please contact the team on private channels before pushing the fix if you have any doubts.
```

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+      description: Please copy and paste any relevant log output. This will be automatically formatted into code, so no need for backticks.
+      render: sh
+  - type: checkboxes
+    id: terms
+    attributes:
+      label: Code of Conduct
+      description: By submitting this issue, you agree to follow our [Code of Conduct](https://github.com/ElrondNetwork/elrond-go/blob/main/.github/CODE_OF_CONDUCT.md)
+      options:
+        - label: I agree to follow this project's Code of Conduct
+          required: true
```
Comment (iulianpascalau) [inline .github/ISSUE_TEMPLATE/bug-report.yml:59]: Let's also add the version. Something like: `v1.3.35.0-0-gfc78258fe/go1.17.6/linux-amd64` and the Host: `DO, AWS, Contabo, no VPS (if you run your validator on a physical machine`

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (bogdan-rosianu) [reply .github/CONTRIBUTING.md:19]: done

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (bogdan-rosianu) [reply .github/CONTRIBUTING.md:24]: done

File: .github/CONTRIBUTING.md (hunk 4 above)
Comment (bogdan-rosianu) [reply .github/CONTRIBUTING.md:25]: added

File: .github/CONTRIBUTING.md (hunk 6 above)
Comment (bogdan-rosianu) [reply .github/CONTRIBUTING.md:32]: added

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 7 above)
Comment (bogdan-rosianu) [reply .github/ISSUE_TEMPLATE/bug-report.yml:59]: added

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (bogdan-rosianu) [reply .github/CONTRIBUTING.md:24]: added

File: .github/CONTRIBUTING.md (hunk 5 above)
Comment (bogdan-rosianu) [reply .github/CONTRIBUTING.md:26]: done

**Review by iulianpascalau — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 7 above)
Comment (raduchis) [reply .github/ISSUE_TEMPLATE/bug-report.yml:75]: :+1:

**Review by raduchis — APPROVED** [review]

**Review by tanghel — APPROVED** [review]

**Review by miiu96 — APPROVED** [review]

**Pull request merged**